src/__tests__/waitFor.test.tsx

```diff
@@ -204,6 +204,43 @@ test.each([false, true])(
   }
 );
 
+const blockThread = (timeToBlockThread: number, legacyFakeTimers: boolean) => {
+  jest.useRealTimers();
+  let end = Date.now() + timeToBlockThread;
+
+  while (Date.now() < end) {
+    // do nothing
+  }
+
+  jest.useFakeTimers({ legacyFakeTimers });
+};
+
+test.each([true, false])(
+  'it should not depend on real time when using fake timers (legacyFakeTimers = %s)',
+  async (legacyFakeTimers) => {
+    jest.useFakeTimers({ legacyFakeTimers });
+
+    const mockErrorFn = jest.fn(() => {
+      // Wait 5 seconds so that check time is longer than interval
+      blockThread(5, legacyFakeTimers);
+      throw new Error('test');
+    });
+
+    await expect(
+      async () =>
+        await waitFor(() => mockErrorFn(), {
+          timeout: 6,
+          interval: 2,
```

Comment:
I don't see how using tight intervals could be fragile since we're using fake timers but maybe I'm thinking too much based in the current implementation so I increased it to 20